src/ConstructiveSolidGeometry/VolumePrimitives/VolumePrimitives.jl

```diff
@@ -1,5 +1,25 @@
-include("Cone.jl")
-include("Box.jl")
-include("RegularPrism.jl")
-include("Sphere.jl")
-include("Torus.jl")
+ClosedPrimitive(p::VP) where {VP <:AbstractVolumePrimitive} = VP(p, COT = ClosedPrimitive, origin = p.origin, rotation = p.rotation)
+OpenPrimitive(p::VP) where {VP <: AbstractVolumePrimitive} = VP(p, COT = OpenPrimitive, origin = p.origin, rotation = p.rotation)
+
+rotation(p::AbstractVolumePrimitive) = p.rotation
+origin(p::AbstractVolumePrimitive) = p.origin
+
+scale(p::VP, s::SVector{3, <:Any}) where {VP <: AbstractVolumePrimitive} = VP(p, scaling = s, origin = p.origin, rotation = p.rotation)
+(*)(s::SVector{3, <:Any}, p::AbstractVolumePrimitive) = scale(p, s)
+
+rotate(p::VP, r::AbstractMatrix) where {VP <: AbstractVolumePrimitive} = VP(p, origin = r * p.origin, rotation = r * p.rotation)
+(*)(r::AbstractMatrix, p::AbstractVolumePrimitive) = rotate(p, r)
+
+translate(p::VP, v::CartesianVector) where {VP <: AbstractVolumePrimitive} = VP(p, origin = p.origin + v, rotation = p.rotation)
+(+)(p::AbstractVolumePrimitive, v::CartesianVector) = translate(p, v)
```

Comment:
Same here, this one ignores `scaling`..

Reply:
If you rotate/translate an object you don't want to change its scale.